picard/mbjson.py

```diff
@@ -143,69 +143,96 @@ def _relation_attributes(relation):
+        name = 'performer:' + _parse_attributes(attribs, reltype, attr_credits)
+    elif reltype == 'mix-DJ' and attribs:
+        if not hasattr(m, "_djmix_ars"):
+            m._djmix_ars = {}
+        for attr in attribs:
+            m._djmix_ars.setdefault(attr.split()[1], []).append(value)
+        return
+    else:
+        try:
+            name = _artist_rel_types[reltype]
+        except KeyError:
+            return
+
+    if context['instrumental'] and name == 'lyricist':
+        return
+
+    m.add_unique(name, value)
+    if name == 'composer':
+        m.add_unique('composersort', valuesort)
+    elif name == 'lyricist':
+        m.add_unique('~lyricistsort', valuesort)
+    elif name == 'writer':
+        m.add_unique('~writersort', valuesort)
+
+
+def _relations_to_metadata_type_work(relation, m, **context):
+    if relation['type'] == 'performance':
+        performance_attributes = _relation_attributes(relation)
+        for attribute in performance_attributes:
+            m.add_unique("~performance_attributes", attribute)
+        work_to_metadata(relation['work'], m, instrumental='instrumental' in performance_attributes)
+
+
+def _relations_to_metadata_type_url(relation, m, **context):
+    if relation['type'] == 'amazon asin' and 'asin' not in m:
+        amz = parse_amazon_url(relation['url']['resource'])
+        if amz is not None:
+            m['asin'] = amz['asin']
+    elif relation['type'] == 'license':
+        m.add('license', relation['url']['resource'])
+
+
+def _relations_to_metadata_type_series(relation, m, **context):
+    if relation['type'] == 'part of':
+        series = relation['series']
+        entity = context['entity']
+        var_prefix = f'~{entity}_' if entity else '~'
+        m.add(f'{var_prefix}series', series['name'])
+        m.add(f'{var_prefix}seriesid', series['id'])
+        m.add(f'{var_prefix}seriescomment', series['disambiguation'])
+        m.add(f'{var_prefix}seriesnumber', relation['attribute-values'].get('number', ''))
+
+
 def _relations_to_metadata(relations, m, instrumental=False, config=None, entity=None):
     config = config or get_config()
-    use_credited_as = not config.setting['standardize_artists']
-    use_instrument_credits = not config.setting['standardize_instruments']
+    context = {
+        'config': config,
+        'instrumental': instrumental,
+        'use_credited_as': not config.setting['standardize_artists'],
+        'use_instrument_credits': not config.setting['standardize_instruments'],
+        'entity': entity,
+    }
+    # build a list of callable methods according to the relation target type
+    # those methods have to start with the same prefix and use same arguments
+    prefix = '_relations_to_metadata_type_'
+    # below the key is the target type, the value the actual function to call
+    funcs = {
+        name[len(prefix):]: v
+        for name, v in globals().items()
+        if name.startswith(prefix) and callable(v)
+    }
```

Comment:
`funcs` does not change, but `_relations_to_metadata`  is called quite frequently. I think we could rather have a dict outside the function in the module. I even think it would be more readable to have an explicit, hard coded mapping of target-type name to function.